bridge/bridgeapp/views.py

```diff
@@ -123,3 +124,28 @@ def post(self, request, thread_id, resp_id):
                 Response.objects.filter(id=resp_id).delete()
         # redirect to single thread page
         return redirect('thread', thread_id, 0)
+
+
+def register(request):
+    if request.method == 'POST':
+        form = UserCreationForm(request.POST)
+
+        if form.is_valid():
+            form.save()
+            username = form.cleaned_data['username']
+            password = form.cleaned_data['password1']
+            user = authenticate(username=username, password=password)
+            login(request, user)
+            return redirect('home')
+
+    else:
+        form = UserCreationForm()
+
+    context = {'form': form}
+    return render(request, 'registration/register.html', context)
+
+
+def logout(request):
```

Comment:
A logout view and template is redundant besides any other concerns. I could not see the use case in any of my imaginations--I might have seen too little to understand your logic, though.

While it might really have some place in some sophisticated website in which the user is redriected to other than the login or home page upon logging out, that certainly is not the case for our simple 3-pager.